src/App.tsx

```diff
@@ -1,10 +1,120 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+
+enum Filter {
+  all = 'All',
+  active = 'Active',
+  completed = 'Completed',
+}
 
 export const App: React.FC = () => {
+  const [value, setValue] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [counter, setCounter] = useState<number>(0);
+  const [filter, setFilter] = useState<string>(Filter.all);
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filter) {
+        case Filter.all:
+          return todo;
+        case Filter.active:
+          return !todo.completed;
+        case Filter.completed:
+          return todo.completed;
+        default:
+          return todo;
+      }
+    });
+  }, [filter, todos]);
+
+  const [error, setError] = useState<string>('');
+
+  useEffect(() => {
+    setTimeout(() => {
+      getTodos()
+        .then(_todos => setTodos(_todos))
+        .catch(() => {
+          setError('Unable to load todos');
+
+          setTimeout(() => setError(''), 3000);
+        });
+    }, 100);
+  }, []);
+
+  useEffect(() => {
+    let counterOfActive: number = 0;
+
+    if (todos.length > 0) {
+      todos.forEach(todo => {
+        if (!todo.completed) {
+          counterOfActive++;
+        }
+      });
+    }
+
+    setCounter(counterOfActive);
+  }, [todos]);
```

Comment:
simplify it using the reduce or filter without the state

Reply:
but I have to update the counter, don't I?

Reply:
```suggestion  
  const nrOfActiveTodos = todos.filter(t => !t.completed).length;
```